### PR TITLE
Allow no newlines before targets

### LINKED ISSUE #538: Error if no base and no blank line
The following earthfile produces an error:
```
build:
    FROM golang:1.13-alpine3.11
    WORKDIR /demo
    COPY go.mod go.sum main.go ./
    RUN go build -o demo .
    ENTRYPOINT ["/demo/demo"]
    SAVE IMAGE example:latest
```
If a blank line is added at the top of the file, the error goes away.

### PR DESCRIPTION
**Pull request by dchw**
For Earthfiles without any `+base`; a (newline/comment) was required before the first target. This eliminates that requirement. This is now valid:

```
no-artifact:
    FROM alpine
    WORKDIR /usr/build
    RUN touch hi
    SAVE ARTIFACT hi
```

Fixes #538

### CODE REVIEW

**Review by alexcb — APPROVED** [review]
looks like an easy fix

File: earthfile2llb/parser/EarthParser.g4 [hunk 1]
```diff
@@ -4,7 +4,7 @@ options {
 	tokenVocab = EarthLexer;
 }
 
-earthFile: NL* stmts? NL+ targets? NL* EOF;
+earthFile: NL* stmts? NL* targets? NL* EOF;
```
Comment (vladaionescu) [inline earthfile2llb/parser/EarthParser.g4:7]: I think we still need to force a NL if base `stmts` are present.

```suggestion
earthFile: NL* (stmts NL)? NL* targets? NL* EOF;
```

**Comment by dchw** [comment]
@alexcb The hard part was ANTLR. Never used it before.

**Review by vladaionescu — COMMENTED** [review]
Good find

File: earthfile2llb/parser/EarthParser.g4 (hunk 1 above)
Comment (dchw) [reply earthfile2llb/parser/EarthParser.g4:7]: This is to enforce a new line between `+base` and `+my-target` right? Just checking for my own understanding

```
FROM xyz
RUN foo

my-target:
    RUN do-it.sh
```

Comment (vladaionescu) [reply earthfile2llb/parser/EarthParser.g4]: not quite.. it's to prevent this (no new line at all):

```
FROM xyz
RUN foo my-target:
    RUN do-it.sh
```

**Comment by vladaionescu** [comment]
Do you want to also add a super simple integration test? Take a look at `examples/tests/Earthfile` and add one more.

File: earthfile2llb/parser/EarthParser.g4 (hunk 1 above)
Comment (dchw) [reply earthfile2llb/parser/EarthParser.g4]: Hm I tried that with the old way and it still failed. Willing to go along with it though

**Comment by dchw** [comment]
> Do you want to also add a super simple integration test? Take a look at `examples/tests/Earthfile` and add one more.

Sure I can add another

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**